core/src/replay_stage.rs

```diff
@@ -811,25 +863,11 @@ impl ReplayStage {
             .filter(|s| s.is_recent && !s.has_voted && !s.vote_threshold)
             .count();
 
-        let mut candidates: Vec<_> = frozen_banks
-            .iter()
-            .zip(stats.iter())
-            .filter(|(_, stats)| stats.is_recent && !stats.has_voted)
-            .collect();
+        let mut candidates: Vec<_> = frozen_banks.iter().zip(stats.iter()).collect();
 
         //highest weight, lowest slot first
-        candidates.sort_by_key(|b| (b.1.weight, 0i64 - b.1.slot as i64));
-
-        candidates.iter().for_each(|(_, stats)| {
-            let mut parents: Vec<_> = if let Some(set) = ancestors.get(&stats.slot) {
-                set.iter().collect()
-            } else {
-                vec![]
-            };
-            parents.sort();
-            debug!("{}: {:?} {:?}", stats.slot, stats, parents,);
-        });
-        let rv = Self::pick_best_fork(ancestors, &candidates);
+        candidates.sort_by_key(|b| (b.1.fork_weight, 0i64 - b.1.slot as i64));
```

Comment:
Can we add a few tests here to illustrate the expected behavior in scenarios where parent-child fork weights are the same? If we are sure that parent weight > child weight in all scenarios, can we assert above that `bank_weight + fork_weight > fork_weight`, i.e. `bank_weight > 0` ( I thnk this is true b/c we always at last simulate one vote)

Reply:
https://github.com/solana-labs/solana/issues/7085